src/wp-admin/network/users.php

```diff
@@ -91,11 +91,23 @@
 								}
 
 								$userfunction = 'all_spam';
-								$blogs        = get_blogs_of_user( $user_id, true );
 
-								foreach ( (array) $blogs as $details ) {
-									if ( ! is_main_site( $details->userblog_id ) ) { // Main site is not a spam!
-										update_blog_status( $details->userblog_id, 'spam', '1' );
+								/**
+								 * Filters whether the blog status for all of a user's blogs should be updated.
+								 *
+								 * @since 6.9.0
+								 *
+								 * @param bool $update_blog_status Whether to update the blog status. Default false.
+								 * @param int $user_id User ID
+								 * @param string $doaction The selected action to execute.
```

Comment:
The `$doaction` argument is not needed because it will always be `spam`:

```suggestion
								 * @param bool $propagate Whether to propagate the blog status. Default false.
								 * @param int  $user_id   User ID.
```